ggml.c

```diff
@@ -1122,13 +1122,37 @@ void dequantize_row_q4_1(const void * restrict x, float * restrict y, int k) {
 #define GGML_F16_EPR  8
 
 // F16 arithmetic is not supported by AVX, so we use F32 instead
-// we take advantage of the _mm256_cvt intrinsics to convert F16 <-> F32
 
 #define GGML_F32Cx8             __m256
 #define GGML_F32Cx8_ZERO        _mm256_setzero_ps()
 #define GGML_F32Cx8_SET1(x)     _mm256_set1_ps(x)
+
+#if defined(__F16C__)
+// the  _mm256_cvt intrinsics require F16C
 #define GGML_F32Cx8_LOAD(x)     _mm256_cvtph_ps(_mm_loadu_si128((__m128i *)(x)))
 #define GGML_F32Cx8_STORE(x, y) _mm_storeu_si128((__m128i *)(x), _mm256_cvtps_ph(y, 0))
+#else
```

Comment:
This should probably be
```c
#elif defined(__SSE3__)
```
to match F32Cx4 counterparts.

Yeah it doesn't compile then if none of them is defined but it's probably intended as so.

Reply:
Other than that, everything looks good to me and after that change we should go ahead and merge.

Reply:
None of this depends on SSE3 anyway, I mindlessly used the same function names as in the SSE3 path for no good reason.

Reply:
The function names should me more accurate to what they actually are now.